Library/ViewModels/CommentCellViewModel.swift

```diff
@@ -190,9 +190,14 @@ public final class CommentCellViewModel:
   public var outputs: CommentCellViewModelOutputs { self }
 }
 
-private func replyButtonHidden(isLoggedOut: Bool, isNotABacker: Bool) -> Bool {
+private func userIsBackingCreatorOrCollaborator(_ project: Project) -> Bool {
+  return (project.personalization.backing != nil || project.personalization.isBacking == .some(true)) ||
+    !project.memberData.permissions.isEmpty
```

Comment:
Looking at the enum for `Permission` I don't see anything specifically suggesting creator or collaborator. How does this work?

Reply:
When you fetch projects, each project comes with permissions relating to the current user.

Reply:
I understand but I am unclear what specifically about this code you've written represents a `creator` or `collaborator`.

Reply:
So on a project MemberData is tied to current logged in user's token, i.e the data for MemberData is personalized to the user.

For example 
This is a snippet response when I logged in with my account to view project "The Project"

```
{
	"id": 1641211140,
	"photo": {
		"key": "assets/033/406/309/9c8d4de5de265da3db0b4fc4c949a557_original.jpg"
	},
	"friends": [],
	"is_starred": false,
	"is_backing": false,
	"permissions": [],
}
```

This is the snippet when I login with Paulo's details who is the creator of  "The Project"

```
{
	"id": 1641211140,
	"photo": {
		"key": "assets/033/406/309/9c8d4de5de265da3db0b4fc4c949a557_original.jpg",
	},
	"friends": [],
	"is_starred": false,
	"is_backing": false,
	"permissions": ["edit_project", "edit_faq", "post", "comment", "view_pledges", "fulfillment"],
}
```

Reply:
@singhhari yeah a valid observation but it seems that in the past all that was done to determine whether the user is a creator or collaborator was to check if this array is empty or not 🤷‍♂️ 

Example: https://github.com/kickstarter/ios-oss/blob/main/Library/ViewModels/MessagesViewModel.swift#L146